xarray/core/dataarray.py

```diff
@@ -3430,6 +3432,225 @@ def pad(
+                indx,
+            ]
+
+        # Promote to a dtype that can handle NaN values if needed.
+        newdtype, fill_value = dtypes.maybe_promote(res.dtype)
+        if promote and newdtype != res.dtype:
+            res = res.astype(newdtype)
+
+        # Put the NaN values back in after removing them, if necessary.
+        if hasna and allna is not None:
+            res = res.where(~allna, fill_value)
+
+        # The dim is gone but we need to remove the corresponding coordinate.
+        del res.coords[dim]
+
+        # Put the attrs back in if needed
+        if keep_attrs:
+            res.attrs = self.attrs
+
+        return res
+
+    def idxmin(
+        self,
+        dim: Optional[Hashable] = None,
+        axis: Optional[int] = None,
+        skipna: Optional[bool] = None,
+        promote: Optional[bool] = None,
+        keep_attrs: Optional[bool] = False,
+        **kwargs: Any,
+    ) -> "DataArray":
+        """Return the coordinate of the minimum value along a dimension.
+
+        Returns a new DataArray named after the dimension with the values of
+        the coordinate along that dimension corresponding to minimum value
+        along that dimension.
+
+        In comparison to `argmin`, this returns the coordinate while `argmin`
+        returns the index.
+
+        Parameters
+        ----------
+        dim : str (optional)
+            Dimension over which to apply `idxmin`.
+        axis : int (optional)
+            Axis(es) over which to repeatedly apply `idxmin`. Exactly one of
+            the 'dim' and 'axis' arguments must be supplied.
+        skipna : bool, optional
+            If True, skip missing values (as marked by NaN). By default, only
+            skips missing values for float dtypes; other dtypes either do not
+            have a sentinel missing value (int) or skipna=True has not been
+            implemented (object, datetime64 or timedelta64).{min_count_docs}
+        promote :  bool (optional)
+            If True (default) dtypes that do not support NaN values will be
+            automatically promoted to those that do.  If False a NaN in the
+            results will raise a TypeError.  If None the result will only be
+            promoted if a NaN is actually present.
+        keep_attrs : bool, optional
+            If True, the attributes (`attrs`) will be copied from the original
+            object to the new one.  If False (default), the new object will be
+            returned without attributes.
+        **kwargs : dict
+            Additional keyword arguments passed on to the appropriate array
+            function for calculating `{name}` on this object's data.
+
+        Returns
+        -------
+        reduced : DataArray
+            New DataArray object with `idxmin` applied to its data and the
+            indicated dimension removed.
+
+        See also
+        --------
+        Dataset.idxmin, DataArray.idxmax, DataArray.min, DataArray.argmin
+        """
```

Comment:
No obligation, but to the extent you're up for it, a couple of examples in the docstrings would be awesome.

Reply:
Done